src/contracts/transfer-gateway.ts

```diff
@@ -66,8 +67,13 @@ export class TransferGateway extends Contract {
   static readonly tokenWithdrawalSignedEventTopic: String = 'event:TokenWithdrawalSigned'
   static readonly contractMappingConfirmedEventTopic: String = 'event:ContractMappingConfirmed'
 
-  static async createAsync(client: Client, callerAddr: Address): Promise<TransferGateway> {
-    const contractAddr = await client.getContractAddressAsync('gateway')
+  static async createAsync(
+    client: Client,
+    callerAddr: Address,
+    loom: boolean = false
```

Comment:
Ugh, this is ugly, can you move this to a new `LoomTransferGateway` class that extends `TransferGateway`?